src/Engines/MeiliSearchEngine.php

```diff
@@ -51,44 +51,56 @@ public function update($models)
             return;
         }
 
-        $index = $this->meilisearch->index($models->first()->searchableAs());
+        $groupedByIndex = $models->groupBy(function ($model) {
+            return $model->searchableAs();
+        });
 
-        if ($this->usesSoftDelete($models->first()) && $this->softDelete) {
-            $models->each->pushSoftDeleteMetadata();
-        }
+        foreach ($groupedByIndex as $index => $collection) {
```

Comment:
I think this could be written as 

```php
$models->groupBy(function ($model) {
            return $model->searchableAs();
})->each(function ($collection, $index) {
    //...
});
```